### ISSUE TITLE
 Error "Impossible to reach SonarQube instance" when try to generate CNES report

### ISSUE THREAD

**Issue by andresv17** [cnescatlab/sonar-cnes-report#155]
#### Actual behavior

When I try to generate the CNES report from the web page I receive the error. This happens with sonarQube 7.9.1 and the version 3.2.2 of cnesreport plugin.

{"errors":[{"msg":"An error has occurred. Please contact your administrator"}]}

#### To reproduce

Click on more --> CNES report
Select the project from the drop down and click on generate

#### Screenshots & log

The web.log shows the following error:

_2020.07.08 15:09:04 ERROR web[AXMrl8X7zBc8glw0AAIx][o.s.s.w.Engine] Fail to process request http://xxxx-api/cnesreport/report?key=xxxxx%3A&author=Administrator&token=5bc06227682bd5371a105b790e395d98d5&generation=Generate
fr.cnes.sonar.report.exceptions.SonarQubeException: Impossible to reach SonarQube instance._

And the main process log this one

Jul 07, 2020 11:25:04 PM fr.cnes.sonar.report.ReportCommandLine execute
INFO: SonarQube URL: https://xxxx.com
Jul 07, 2020 11:25:34 PM fr.cnes.sonar.report.providers.SonarQubeInfoProvider getSonarQubeStatus
SEVERE: Impossible to reach SonarQube instance.
fr.cnes.sonar.report.exceptions.SonarQubeException: Impossible to reach SonarQube instance.

Thanks in advance.

**Comment by Sancretor** [comment]
Hi @andresv17 

Thanks for your feedback !

In the app log file, you can find these lines : 
```
Caused by: java.lang.IllegalStateException: Fail to request https://sirius-sonar.fonyou.com/api/system/status
...
Caused by: java.net.SocketTimeoutException: connect timed out
```

Before the plugin produces the report, the plugin webpage tries to reach the SonarQube server to check if the API are available.  
Unfortunately it seems it cannot reach it, so you should investigate on this to solve your issue.

**Comment by andresv17** [comment]
Hi @Sancretor 

It was one of the first things I checked. Where I try to reach the URL from the webpage it works.
I receive this message 

{"id":"A197EC53-AXNJKzdhy9FnjxZNbKnk","version":"8.2.0.32929","status":"UP"}

If I make a curl from the server where I have the sonar and the plugin the result is the same. I really don't know why this happens just when I try to generate the report

**Comment by Sancretor** [comment]
Hi @andresv17 

In your first message you mentioned a SonarQube 7.9.1, and now your message shows a 8.2.0 ...
I just want to be sure you are testing the same thing.

If a curl from the server works the standalone tool will work, no doubt about it.
You tried to reach "https://sirius-sonar.fonyou.com/api/system/status" from the same browser/computer you failed to generate the report from the plugin page ? And it worked ?
In this case, I don't know what can cause this issue but it is not related to the plugin...

**Comment by phreak1990** [comment]
Hi @Sancretor 

I am also facing the same issue with sonarqube.
SonarQube version: 7.9.4
sonar-cnes-report-3.2.2.jar


2020.08.08 18:32:04 ERROR web[AXPMhYIgN938Mc0TAADo][o.s.s.w.WebServiceEngine] Fail to process request http://localhost:9000/api/cnesreport/report?key=test_repo&branch=master&author=Administrator&token=<token>&generation=Generate
fr.cnes.sonar.report.exceptions.SonarQubeException: Impossible to reach SonarQube instance.


root@test:/home/test/Documents/sonarqube/extensions/plugins# java -jar sonar-cnes-report-3.2.2.jar -p test_repo
SonarQube URL: http://localhost:9000
SonarQube online: true
Detected SonarQube version: 7.9.4
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.poi.openxml4j.util.ZipSecureFile$1 (file:/home/test/Documents/sonarqube/extensions/plugins/sonar-cnes-report-3.2.2.jar) to field java.io.FilterInputStream.in
WARNING: Please consider reporting this to the maintainers of org.apache.poi.openxml4j.util.ZipSecureFile$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Report generation: SUCCESS

**Comment by makhaly** [comment]
I am facing the same issue after upgrading from SonarQube v8.2 Developer to v8.4.

**Comment by Sancretor** [comment]
Just to let you know that we just released a new version compatible with 8.9 LTS.
We completely changed the way the plugin handles API calls.
It now uses internal SonarQube classes instead of localhost http calls, so it may resolve your issue.

Please let us know if it doesn't.
